File: stone_processor/src/main/java/com/github/klee0kai/stone/exceptions/RecursiveProviding.java
package com.github.klee0kai.stone.exceptions;

import javax.lang.model.element.Element;

public class RecursiveProviding extends StoneException {

    public RecursiveProviding(String message) {
        super(message, null);
    }

    public RecursiveProviding(String message, Throwable cause) {
        super(message, cause);
    }

    public RecursiveProviding(String message, Throwable cause, Element element) {
        super(message, cause,element);
    }

}

File: stone_ksp/src/commonMain/kotlin/com/github/klee0kai/thekey/stone/ksp/exceptions/RecursiveProviding.kt
package com.github.klee0kai.thekey.stone.ksp.exceptions

import com.google.devtools.ksp.symbol.KSNode

class RecursiveProviding(
    message: String? = null,
    cause: Throwable? = null,
    element: KSNode? = null,
) : StoneException(
    message = message,
    cause = cause,
    element = element,
)